crates/rpc-types-engine/src/attributes.rs

```diff
@@ -52,6 +55,46 @@ impl OpPayloadAttributes {
     pub fn decode_eip_1559_params(&self) -> Option<(u32, u32)> {
         self.eip_1559_params.map(decode_eip_1559_params)
     }
+    /// Returns an iterator over `WithEncoded<OpTxEnvelope>`
+    pub fn decoded_transactions_with_encoded(
+        &self,
+    ) -> impl Iterator<Item = WithEncoded<OpTxEnvelope>> + '_ {
+        self.transactions.iter().flatten().filter_map(|tx_bytes| {
+            OpTxEnvelope::decode_2718(&mut tx_bytes.as_ref())
+                .ok()
+                .and_then(|env| OpTxEnvelope::try_from(env).ok())
+                .map(|env| WithEncoded::new(tx_bytes.clone(), env))
+        })
+    }
+    /// Returns an iterator over `WithEncoded<Recovered<OpTxEnvelope>>`
+    pub fn recovered_transactions_with_encoded(
+        &self,
+    ) -> impl Iterator<Item = WithEncoded<Recovered<OpTxEnvelope>>> + '_ {
+        self.transactions.iter().flatten().filter_map(|tx_bytes| {
+            let env = OpTxEnvelope::decode_2718(&mut tx_bytes.as_ref()).ok()?;
+            let txenv = env.try_into_eth_envelope().ok()?;
+            let recovered = txenv.try_into_recovered().ok()?;
+            // Convert EthereumTxEnvelope back to OpTxEnvelope
+            match OpTxEnvelope::try_from_eth_envelope(recovered.inner().clone()) {
+                Ok(op_tx) => Some(WithEncoded::new(tx_bytes.clone(), Recovered::new_unchecked(op_tx, recovered.signer().clone()))),
+                Err(_) => None,
+            }
+        })
```

Comment:
ideally we can reuse a single function that does the encoding and then simply layer the things we need on top, like sender recovery